src/util/codec/number.rs

```diff
@@ -219,6 +222,150 @@ pub trait NumberDecoder: Read {
 
 impl<T: Read> NumberDecoder for T {}
 
+type Bytes<'a> = &'a [u8];
```

Comment:
I prefer not to define this type alias. The name `Bytes` gives a false implication about whether it is a reference or a value. `&[u8]` is much more cleaner.

Reply:
I think this alias will be much easier and clear, otherwise, we need to define the lifecycle in every decoding function.